metadata_backend/helpers/metax_api_handler.py

```diff
@@ -0,0 +1,261 @@
+            status = resp.status
+            if status == 204:
+                LOG.info(f"Deleted draft dataset {metax_id} from Metax service")
+            else:
+                # TODO: how front end should react on this??
+                reason = await resp.text()
+                raise self.process_error(status, reason)
+
+    async def publish_dataset(self, folder_id: str) -> None:
+        """Publish draft dataset to Metax service.
+
+        Fetch metadataObjects for published folder. Publish each object within Metax service and
+        update object's Metax status to db.
+
+        :param folder_id: Folder ID where metadata objects to publish resides
+        """
+        folder = await FolderOperator(self.db_client).read_folder(folder_id)
+        operator = Operator(self.db_client)
+        for object in folder["metadataObjects"]:
+            if object["schema"] in {"study", "dataset"}:
+                data, _ = await operator.read_metadata_object(object["schema"], object["accessionId"])
+                if isinstance(data, dict):
+                    metax_id = data["metaxIdentifier"]
+                    doi = data["doi"]
+                async with aiohttp.ClientSession() as sess:
+                    resp = await sess.post(
+                        f"{self.metax_url}{self.publish_route}",
+                        params={"identifier": metax_id},
+                        auth=aiohttp.BasicAuth(self.username, self.password),
+                    )
+                    status = resp.status
+                    if status == 200:
+                        preferred_id = await resp.json()
+                        if doi != preferred_id["preferred_identifier"]:
+                            LOG.warning(
+                                f"Metax Preferred Identifier {preferred_id['preferred_identifier']} "
+                                f"does not match object's DOI {doi}"
+                            )
+                        LOG.debug(
+                            f"Object {object['schema']} with accession ID {object['accessionId']} is "
+                            "published to Metax service."
+                        )
+                        # This must be updated as Metax identifier will be moved to folder from object after publishing
+                        # await operator.update_metadata_object(
+                        #     object["schema"],
+                        #     object["accessionId"],
+                        #     {
+                        #         "metaxIdentifier": {
+                        #             "identifier": metax_id,
+                        #             "status": "published",
+                        #         }
+                        #     },
+                        # )
+                    else:
+                        # TODO: how front end should react on this??
+                        reason = await resp.text()
+                        raise self.process_error(status, reason)
```

Comment:
@hannyle @saulipurhonen

Reply:
this is already handled in the middleware with the redirect to the error pages, not sure why we need to do more here than raising the error. See: https://github.com/CSCfi/metadata-submitter/blob/develop/metadata_backend/api/middlewares.py#L18-L71